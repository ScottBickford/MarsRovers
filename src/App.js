import React from "react";
import "./App.css";
import { Container } from "semantic-ui-react";
import RoverInstructionsPage from "./features/roverInstructions/RoverInstructionsPage";

function App() {
  return (
    <Container className='main'>
      <h2>Rover Instructions</h2>
      <RoverInstructionsPage />
    </Container>
  );
}

export default App;
