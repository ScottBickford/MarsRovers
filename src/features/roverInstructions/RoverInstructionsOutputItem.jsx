import React, { Fragment } from 'react';

const RoverInstructionsOutputItem = ({ rover : {positions} }) => {
  const lastPosition = positions && positions[positions.length - 1];
  return (
    <Fragment>
      {lastPosition.error && (
        <div style={{ color: 'red' }}>{lastPosition.error}</div>
      )}
      {!lastPosition.error && (
        <div>
          {lastPosition.x} {lastPosition.y} {lastPosition.compassPoint}
        </div>
      )}
    </Fragment>
  );
};

export default RoverInstructionsOutputItem;
