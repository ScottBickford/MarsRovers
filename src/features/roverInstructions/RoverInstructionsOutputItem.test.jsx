import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import RoverInstructionsOutputItem from './RoverInstructionsOutputItem';
import { roverMovementsTestObject } from '../../App.test';

describe('RoverInstructionsOutputItem', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<RoverInstructionsOutputItem rover={roverMovementsTestObject.rovers[0]} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes all props to Input', () => {
    const { getByText } = render(<RoverInstructionsOutputItem rover={roverMovementsTestObject.rovers[0]} />);
    const buttonElement = getByText(/1 3 N/i);
    expect(buttonElement).toBeInTheDocument();
  });
});