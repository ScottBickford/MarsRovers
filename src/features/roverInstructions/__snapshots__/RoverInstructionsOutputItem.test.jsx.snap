// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoverInstructionsOutputItem snapshot renders 1`] = `
<div>
  1
   
  3
   
  N
</div>
`;
